runtime/compiler/net/MessageBuffer.cpp

```diff
@@ -21,18 +21,50 @@
  *******************************************************************************/
 
 #include "net/MessageBuffer.hpp"
+#include "infra/CriticalSection.hpp"
 #include <cstring>
 
 namespace JITServer
 {
+
+TR::Monitor *MessageBuffer::_totalBuffersMonitor = NULL;
+int MessageBuffer::_totalBuffers = 0;
+TR::PersistentAllocator *MessageBuffer::_allocator = NULL;
+
 MessageBuffer::MessageBuffer() :
-   _capacity(INITIAL_BUFFER_SIZE),
-   _allocator(TR::Compiler->persistentGlobalAllocator())
+   _capacity(INITIAL_BUFFER_SIZE)
    {
+   OMR::CriticalSection cs(getTotalBuffersMonitor());
+
+   if (!_allocator)
+      {
+      if (J9::PersistentInfo::_remoteCompilationMode == JITServer::CLIENT)
+         {
+         TR::PersistentAllocatorKit kit(1 << 20/*1 MB*/, *TR::Compiler->javaVM);
```

Comment:
I think you must ensure that we use mmap for the allocations:
```
uint32_t memoryType = MEMORY_TYPE_VIRTUAL; // Force the usage of mmap for allocation
TR::PersistentAllocatorKit kit(1 << 20/*1 MB*/, *TR::Compiler->javaVM, memoryType);
```

Reply:
Oh, allocations aren't guaranteed to be anonymous private mappings unless they are of a certain size in glibc. Probably better to be on the safe side. Fixed.